flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/BufferBuilderAndConsumerTest.java

```diff
@@ -202,22 +202,30 @@ private static void testIsFinished(int writes) {
 		for (int i = 0; i < writes; i++) {
 			assertEquals(Integer.BYTES, bufferBuilder.appendAndCommit(toByteBuffer(42)));
 		}
+		int expectedWrittenBytes = writes * Integer.BYTES;
 
 		assertFalse(bufferBuilder.isFinished());
 		assertFalse(bufferConsumer.isFinished());
+		assertEquals(0, bufferConsumer.getWrittenBytes());
 
 		bufferConsumer.build();
-
 		assertFalse(bufferBuilder.isFinished());
 		assertFalse(bufferConsumer.isFinished());
+		assertEquals(expectedWrittenBytes, bufferConsumer.getWrittenBytes());
 
-		bufferBuilder.finish();
-
+		int actualWrittenBytes = bufferBuilder.finish();
+		assertEquals(expectedWrittenBytes, actualWrittenBytes);
 		assertTrue(bufferBuilder.isFinished());
 		assertFalse(bufferConsumer.isFinished());
+		assertEquals(expectedWrittenBytes, bufferConsumer.getWrittenBytes());
```

Comment:
nit: why not also verify `bufferBuilder.getWrittenBytes()` (at all locations where this is verified)?